use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,491 @@
+is `37c8` and virtual function device ID is `37c9`. The following command checks 
+if VF's are enabled for any of the currently known Intel QAT device ID's. 
+
+```
+$ for i in 0442 0443 37c9 19e3; do lspci -d 8086:$i; done
+```
+
+## Prepare Kata Containers
+
+### Download Kata Kernel Source
+
+This follows the instructions from 
+[https://github.com/kata-containers/packaging/tree/master/kernel](https://
+github.com/kata-containers/packaging/tree/master/kernel). The config for the 
+kata-kernel is at 
+[https://github.com/kata-containers/packaging/tree/master/kernel/configs](
+https://github.com/kata-containers/packaging/tree/master/kernel/configs) In 
+this example, the latest kernel supported by Kata is automatically used.
+There are some patches that must be installed as well which the 
+build-kernel.sh script should automatically apply, but if you are using a 
+different kernel version then they may need to be manually applied. Since the 
+Kata Containers kernel has a minimal set of kernel flags set, the config must 
+be updated to set some of the `CRYPTO` and `STACKPROTECTOR` flags to enabled. 
+This may change with different kernel versions. The following was tested with 
+kernel `v4.19.28-41`.
+
+```
+$ mkdir -p $GOPATH
+$ cd $GOPATH
+$ go get -v github.com/kata-containers/packaging
+$ $GOPATH/src/github.com/kata-containers/packaging/kernel/build-kernel.sh setup
+```
+
+### Build Kata Kernel
+
+```
+$ export LINUX_VER=$(ls -d kata*)
+$ sed -i 's/EXTRAVERSION =/EXTRAVERSION = .qat.container/' $LINUX_VER/Makefile
+
+# Change kernel config options to support QAT
+$ sed -i 's/# CONFIG_PCI_IOV is not set/CONFIG_PCI_IOV=Y/' $LINUX_VER/.config
+$ sed -i 's/# CONFIG_CRYPTO_SHA1 is not set/CONFIG_CRYPTO_SHA1=Y/' $LINUX_VER/.config
+$ sed -i 's/# CONFIG_CRYPTO_DH is not set/CONFIG_CRYPTO_DH=Y/' $LINUX_VER/.config
+$ sed -i 's/# CONFIG_STACKPROTECTOR is not set/CONFIG_STACKPROTECTOR=Y/' $LINUX_VER/.config
+$ $GOPATH/src/github.com/kata-containers/packaging/kernel/build-kernel.sh build
+```
+
+__Note:__ 
+There may be additional kernel parameters that need to be set to yes when 
+compiling.
+
+### Copy Kata Kernel
+
+```
+$ mkdir -p $KATA_KERNEL_LOCATION
+$ cp $LINUX_VER/arch/x86/boot/bzImage $KATA_KERNEL_LOCATION/vmlinuz-${LINUX_VER}_qat
+```
+
+### Prepare Kata root filesystem
+
+These instructions build upon the OS builder instructions located at 
+[https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#create-and-install-rootfs-and-initrd-image](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#create-and-install-rootfs-and-initrd-image). This example will use Clear Linux (Kata Containers default) as the root filesystem with systemd as the init and will add in the kmod binary which is not a standard binary in a Kata image.  Docker should already be installed. If a custom kata-agent needs to be used please refer to the link above on how to add in.
```

Comment:
You can probably trim this link name down to just `Developer-Guide.md#create-and-install-rootfs-and-initrd-image` or less, and also the actual link itself can I think be made relative, as this doc is in the same repo.

Reply:
I'll go with  `[Developer Guide](../Developer-Guide.md#create-and-install-rootfs-and-initrd-image)`   I hope this link works as i can't test locally until I submit.

Reply:
I should note this link fails the kata-check-markdown tool because it can't follow the path.  I am not 100% sure the best way to reference this but I think I did it the correct way.

Reply:
Hi @eadamsintel - I'm confused. The markdown checker certainly can follow links and the link above does work with it...? I'm talking about:

```
[Developer Guide](../Developer-Guide.md#create-and-install-rootfs-and-initrd-image)
```

Reply:
I assume it is because I was running kata-check-markdown on the file across a network samba share vs having the entire git tree locally. Still, the share included the entire git tree so I can't explain why it was not able to traverse.

Reply:
That sounds like you've been bitten by the `dos2unix` issue again. All the Kata tooling is designed for Linux :smile:

Reply:
Maybe. Here is what it looks like when running on Linux across a Samba Share.
```
eadams@eadams-MOBL /media/samba/documentation/use-cases $ ~/go/src/github.com/kata-containers/tests/cmd/check-markdown/kata-check-markdown check using-Intel-QAT-and-kata.md
INFO[0000] Checking file                                 commit=5f887bee08bb996916dacd76df47d23833564295 file=using-Intel-QAT-and-kata.md name=kata-check-markdown pid=1448 source=check-markdown version=0.0.1
INFO[0000] Checking file                                 commit=5f887bee08bb996916dacd76df47d23833564295 file=../Developer-Guide.md name=kata-check-markdown pid=1448 source=check-markdown version=0.0.1
FATA[0002] file="../Developer-Guide.md": failed to find heading for link "build-and-install-the-kata-containers-runtime" ({Doc:../Developer-Guide.md Address:build-and-install-the-kata-containers-runtime ResolvedPath: Description:Build and install the Kata Containers runtime Type:internal-link})  commit=5f887bee08bb996916dacd76df47d23833564295 name=kata-check-markdown pid=1448 source=check-markdown version=0.0.1
```

As I look at it closer it looks like it is having issues following a link in the Developer Guide.  That link looks valid so I can't explain it.

Reply:
I can only recommend that you run all Kata tooling on Linux, as it was designed for that system :)